File: js/src/formulas/heron.ts
export default function heron (a: number, b: number, c: number) {
  const cosTheta = ((a * a) + (b * b) - (c * c)) / (2 * a * b)
  const sinTheta = Math.sqrt(1 - (cosTheta ** 2))
  return (a * b * sinTheta) / 2
}

File: py/src/formulas/heron.py
import math

def heron(a: float, b: float, c: float) -> float:
    cos_theta = ((a * a) + (b * b) - (c * c)) / (2 * a * b)
    sin_theta = math.sqrt(1 - (cos_theta ** 2))
    return (a * b * sin_theta) / 2
